### PR TITLE
Update to meteor 1.3.4.4

### PR DESCRIPTION
**Pull request by rhettlivingston**
The changes contained in this PR update meteor/todos to Meteor 1.3.4.4 and fix problems in both the application and the ESLint configuration that arose during the update.

Notably, 

- Promise.denodeify was removed by Meteor 1.3.3 and has been replaced by a local implementation
- The ESLint configuration was broken by a very obscure interaction between eslint-plugin-meteor, eslint-plugin-imports, NPM v3.x.x and Node v0.10.x that has been fixed
- allowing the use of conditional imports and exports also drove an ESLint change

Also, the application was changed to utilize conditional imports to replace existing 'require's and a conditional export in a case requiring a client-only interface.

### CODE REVIEW

File: imports/api/generate-data.app-tests.js [hunk 1]
```diff
@@ -25,15 +26,15 @@ Meteor.methods({
   },
 });
 
-let generateData; // eslint-disable-line import/no-mutable-exports
+// let generateData; // eslint-disable-line import/no-mutable-exports
```
Comment (tmeasday) [inline imports/api/generate-data.app-tests.js]: Why is this line still here?

**Comment by tmeasday** [comment]
Looks great!

File: imports/api/generate-data.app-tests.js (hunk 1 above)
Comment (rhettlivingston) [inline imports/api/generate-data.app-tests.js]: Because I'm sometimes sloppy. Fixing.

**Comment by rhettlivingston** [comment]
That's a seriously outdated Node / NPM configuration in the CircleCI setup. Why does it not use `meteor npm install`?

**Comment by tmeasday** [comment]
Good question! I'll fix it now.

**Comment by tmeasday** [comment]
I don't think the node version actually matters at all, I wonder if we can get rid of it. Anyways, feel free to cherrypick / merge https://github.com/meteor/todos/commit/b2fd0af1c37d60700e6365feb1ce95d54c97a036

**Comment by tmeasday** [comment]
Hmm, we might need some more changes, will report back in a sec after circle-reruns on master

**Comment by rhettlivingston** [comment]
Whoops. Didn't see your last note. However, I may pass before you because I've got that change required by lint, hence why we now have a conflict.

**Comment by tmeasday** [comment]
Oh, nice! I'll merge your changes manually

**Comment by tmeasday** [comment]
Awesome!

**Comment by rhettlivingston** [comment]
A bit late, but `meteor npm run test-app` has been running reliably since your last updates. I would think it could be added to the CircleCI configuration too.

**Pull request merged**